peps/pep-0815.rst

```diff
@@ -0,0 +1,2281 @@
+complexity in managing platform-specific dependencies and compatibility.
+That complexity affects the installation process for users and increases
+the maintenance burden for package authors. The ``x86_64`` compiler
+flags further emphasizes the urgent need for a more expressive and
+efficient solution in the Python packaging ecosystem.
+
+As illustrated by
+`archspec <https://github.com/archspec/archspec>`__’s authors and
+maintainers, the ability to optimize a package for a specific
+architecture can lead to significant performance improvement (and often
+also reduce power requirements, etc.).
+
+.. figure:: pep-0815/avx512_gromacs_benchmark.png
+    :alt: A bar graph comparing GROMACS performance (in ns/day) with
+          various targets. The first two bars are labeled "yum (2018.8)"
+          and "generic (SSE2)", reach about 1.0 ns/day and are both
+          marked as "SSE2". The next bar is labeled "ivybridge" ("AVX")
+          and reaches almost 1.5 ns/day.  Two following bars are labeled
+          "haswell" and "broadwell" (both "AVX2") and exceed 1.5 ns/day
+          slightly.  The last two bars are labeled "skylake_avx512" and
+          "cascadelake" (both "AVX512") and reach almost 2.0 ns/day.
+
+    GROMACS AVX512 Benchmark
+
+..
+
+    “Compiling `GROMACS <https://www.gromacs.org/>`__ for architectures
+    that can exploit the AVX-512 instructions supported by the Intel
+    Cascade Lake microarchitecture gives an additional 18% performance
+    improvement relative to using AVX2 instructions, with a speedup of
+    about 70% compared to a generic GROMACS installation with only SSE2.”
+
+    — `archspec: A library for detecting, labeling, and reasoning about
+    microarchitectures <https://tgamblin.github.io/pubs/archspec-canopie-hpc-2020.pdf>`__
+
+This PEP proposes a systematic and scalable approach to selecting
+optimized wheels based on platform characteristics, which will help
+Python’s usage expand across diverse computing environments, from cloud
+computing to embedded systems and AI accelerators.
+
+
+User stories
+------------
+
+- A user wants to install a version of NumPy that is specialized for
+  their CPU architecture.
+
+- A user wants to install a version of PyTorch that is specialized for
+  their GPU architecture.
+
+- A user wants to install a version of mpi4py that has certain features
+  enabled (e.g. specific MPI implementations for their hardware).
+
+- A library maintainer wants to build their library for wasm32-wasi with
+  and without pthreads support.
+
+- A library maintainer wants to build their library for an Emscripten
+  platform for Pyodide with extensions for graphics compiled in.
+
+- A library maintainer wants to provide packages of their game library
+  using different graphics backends.
+
+- SciPy wants to provide packages built against different BLAS
+  libraries, like OpenBLAS and Accelerate on macOS. This is something
+  they indirectly do today
```

Comment:
```suggestion
  they indirectly do today.
```

Reply:
@DEKHTIARJonathan, any chance you wrote that? Or have a clue what this is about?

Reply:
See https://github.com/wheelnext/pep_xxx_wheel_variants/issues/54#issuecomment-3014031815. I don't think "they indirectly do today" is helpful indeed. More like "SciPy already ships wheels linked against different BLAS libraries on macOS, however they are not selectable by the user but keyed on different OS versions."